### PR TITLE
VMIM controller - move unit tests to fake client

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
Use fake client instead of mocking in order to simulate real behavior.

Fixes #

### Why we need it and why it was done in this way
Lower the bar for unit tests, eliminate false positives. The fake is standard for testing.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/watch/migration_test.go [hunk 1]
```diff
@@ -1562,29 +1553,24 @@ var _ = Describe("Migration watcher", func() {
...
-				return true, creation.GetObject(), nil
-			})
 			controller.Execute()
 
 			testutils.ExpectEvent(recorder, SuccessfulCreatePodReason)
+			expectPodCreation(vmi.Namespace, vmi.UID, migration.UID, 1, 0, 1)
+			pods, err := kubeClient.CoreV1().Pods(vmi.Namespace).List(context.Background(), metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s,%s=%s", virtv1.MigrationJobLabel, string(migration.UID), virtv1.CreatedByLabel, string(vmi.UID)),
+			})
+			//TODO FIXME CONTAINS
```
Comment (jean-edouard) [inline pkg/virt-controller/watch/migration_test.go:1563]: What do you mean?

Comment (fossedihelm) [reply pkg/virt-controller/watch/migration_test.go]: Leftover from implementing step.
Thank you, eagle eye :)

**Review by jean-edouard — APPROVED** [review]

**Comment by jean-edouard** [comment]
Needs `make generate && make`

**Comment by fossedihelm** [comment]
> Needs `make generate && make`

Done, thanks :)

**Review by jean-edouard — APPROVED** [review]

**Pull request merged**